khmer/_khmer.cc

```diff
@@ -5064,15 +5090,32 @@ static PyObject * forward_hash_no_rc(PyObject * self, PyObject * args)
         return NULL;
     }
 
-    return PyLong_FromUnsignedLongLong(_hash_forward(kmer, ksize));
+    HashIntoType h(_hash_forward(kmer, ksize));
+    PyObject * hash = _PyLong_FromByteArray(h.bytes.data(), h.bytes.size(),
+                                            0, 0);
+    return hash;
 }
 
 static PyObject * reverse_hash(PyObject * self, PyObject * args)
 {
-    HashIntoType val;
+    PyObject * val;
+    HashIntoType hash;
     WordLength ksize;
 
-    if (!PyArg_ParseTuple(args, "Kb", &val, &ksize)) {
+    if (!PyArg_ParseTuple(args, "Ob", &val, &ksize)) {
+        return NULL;
+    }
+    if (PyLong_Check(val)) {
+        _PyLong_AsByteArray((PyLongObject *)val,
```

Comment:
This seems like a nice place to write a macro - replace _PyLong_AsByteArray(...) with a standard function/macro whatever, that could work for whatever we end up doing.